src/python/WMCore/ReqMgr/Utils/Validation.py

```diff
@@ -43,25 +85,31 @@ def validate_request_update_args(request_args, config, reqmgr_db_service, param)
     """
     # this needs to be deleted for validation
     request_name = request_args.pop("RequestName")
+
     couchurl = '%s/%s' % (config.couch_host, config.couch_reqmgr_db)
     workload = WMWorkloadHelper()
     workload.loadSpecFromCouch(couchurl, request_name)
 
+    request = reqmgr_db_service.getRequestByNames(request_name)
+    request = request[request_name]
+    reqArgsDiff = _validate_request_allowed_args(request, request_args)
+
     # validate the status
-    if "RequestStatus" in request_args:
-        validate_state_transition(reqmgr_db_service, request_name, request_args["RequestStatus"])
-        if request_args["RequestStatus"] in STATES_ALLOW_ONLY_STATE_TRANSITION:
+    if "RequestStatus" in reqArgsDiff:
+        validate_state_transition(reqmgr_db_service, request_name, reqArgsDiff["RequestStatus"])
+        if reqArgsDiff["RequestStatus"] in STATES_ALLOW_ONLY_STATE_TRANSITION:
             # if state change doesn't allow other transition nothing else to validate
             args_only_status = {}
-            args_only_status["RequestStatus"] = request_args["RequestStatus"]
-            args_only_status["cascade"] = request_args.get("cascade", False)
+            args_only_status["RequestStatus"] = reqArgsDiff["RequestStatus"]
+            args_only_status["cascade"] = reqArgsDiff.get("cascade", False)
             return workload, args_only_status
-        elif request_args["RequestStatus"] == 'assigned':
-            workload.validateArgumentForAssignment(request_args)
+        elif reqArgsDiff["RequestStatus"] == 'assigned':
+            workload.validateArgumentForAssignment(reqArgsDiff)
 
-    validate_request_priority(request_args)
+    if 'RequestPriority' in reqArgsDiff:
```

Comment:
This check is already performed inside the function, so we can keep the previous and shorter implementation.

Reply:
If you refer to the `_handleNoStatusUpdate` function, then no, there we check to find out which is the proper set of actions to be taken, here we validate the request priority value sent IIUC.

Reply:
I was referring to `validate_request_priority`, in the line below.

Reply:
pong

Reply:
why would we need to increase  the stack with an extra function call, when we  already know at this very stage that  a `RequestPriority` validation won't be needed. Anyway .. I'll remove it and will let us call the `validate_request_priority` constantly